gremlin-scala/src/test/scala/gremlin/scala/TraversalSpec.scala

```diff
@@ -394,6 +395,34 @@ class TraversalSpec extends WordSpec with Matchers {
     }
   }
 
+  "choose" should {
+    "work for if/then/else semantics" in new Fixture {
+
+      val traversal: GremlinScala[String, _] = graph
+        .V.hasLabel("person")
+        .choose(
+          _.values("age").is(P.lte(30)),
+          _.in(),
```

Comment:
personally I'd use named parameters here, i.e. `trueChoice = _.in()` - just to make it more readable